basic/src/test/java/org/jboss/hal/testsuite/test/configuration/singleton/ElectionPolicyTestCase.java

```diff
@@ -0,0 +1,206 @@
+package org.jboss.hal.testsuite.test.configuration.singleton;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.jboss.arquillian.drone.api.annotation.Drone;
+import org.jboss.arquillian.graphene.page.Page;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.dmr.ModelNode;
+import org.jboss.hal.testsuite.category.Shared;
+import org.jboss.hal.testsuite.creaper.ManagementClientProvider;
+import org.jboss.hal.testsuite.creaper.ResourceVerifier;
+import org.jboss.hal.testsuite.fragment.shared.modal.WizardWindowWithAdvancedSelectBoxOptions;
+import org.jboss.hal.testsuite.page.config.SingletonSubsystemPage;
+import org.jboss.hal.testsuite.util.AvailablePortFinder;
+import org.jboss.hal.testsuite.util.ConfigChecker;
+import org.jboss.hal.testsuite.util.ConfigUtils;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.WebDriver;
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RunWith(Arquillian.class)
+@Category(Shared.class)
+public class ElectionPolicyTestCase {
+
+    @Drone
+    private WebDriver browser;
+
+    @Page
+    private SingletonSubsystemPage page;
+
+    private static final String
+            SINGLETON_POLICY_RESOURCE = "s-pol_" + RandomStringUtils.randomAlphanumeric(5),
+            SINGLETON_POLICY_EDIT_RESOURCE = "s-pol-edit_" + RandomStringUtils.randomAlphanumeric(5),
+            CACHE_CONTAINER_RESOURCE = "cache-container-singleton_" + RandomStringUtils.randomAlphanumeric(5),
+            CACHE_CONTAINER = "cache-container",
+            SINGLETON_POLICY = "singleton-policy",
+            ELECTION_POLICY = "election-policy",
+            NAME_PREFERENCES = "name-preferences",
+            SOCKET_BINDING_PREFERENCES = "socket-binding-preferences",
+            SIMPLE = "simple";
+
+    private static final Address
+            SINGLETON_SUBSYSTEM_ADDRESS = ConfigUtils.isDomain() ? Address.of("profile", "full-ha").and("subsystem", "singleton") : Address.subsystem("singleton"),
+            INFINISPAN_SUBSYSTEM_ADDRESS = ConfigUtils.isDomain() ? Address.of("profile", "full-ha").and("subsystem", "infinispan") : Address.subsystem("infinispan"),
```

Comment:
The check for doamin isn't needed, the prefix for subsystem in domain is done based on management client configuration.